src/eventmaster/api_handlers.go

```diff
@@ -388,31 +388,49 @@ func (h *httpHandler) handleGetDc(w http.ResponseWriter, r *http.Request, _ http
 	h.sendResp(w, "", string(str), r.URL.Path)
 }
 
-func (h *httpHandler) handleGitHubEvent(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
-	var info map[string]interface{}
-
-	defer r.Body.Close()
-	reqBody, err := ioutil.ReadAll(r.Body)
-	if err != nil {
-		h.sendError(w, http.StatusBadRequest, err, "Error reading request body", r.URL.Path)
-		return
-	}
-	if err := json.Unmarshal(reqBody, &info); err != nil {
-		h.sendError(w, http.StatusBadRequest, err, "Error JSON decoding body of request", r.URL.Path)
-		return
-	}
-
-	id, err := h.store.AddEvent(&UnaddedEvent{
-		Dc:        "github",
-		Host:      "github",
-		TopicName: "github",
-		Data:      info,
-	})
-	if err != nil {
-		h.sendError(w, http.StatusInternalServerError, err, "Error adding event to store", r.URL.Path)
-		return
+func (h *httpHandler) handlePluginEvent(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	pluginName := ps.ByName("name")
+	if plugin, ok := Plugins[pluginName]; ok {
+		defer r.Body.Close()
+		rawData, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			h.sendError(w, http.StatusInternalServerError, err, "Error reading request body", r.URL.Path)
+			return
+		}
+		valid := plugin.ValidateAuth(r, rawData)
+		if !valid {
+			h.sendError(w, http.StatusBadRequest, errors.New("Invalid signature"), "Error validating request", r.URL.Path)
+			return
+		}
+		info, err := plugin.ParseRequest(r, rawData)
+		if err != nil {
+			h.sendError(w, http.StatusInternalServerError, err, "Error parsing event", r.URL.Path)
+			return
+		}
+		dc, host, user, targetHostSet, tags, parentEventId, eventTime, data := plugin.ParseInfo(r, info)
+		var ids []string
+		for _, topic := range plugin.GetTopics() {
```

Comment:
The thought here was that the plugin would parse the payload and return a list of events to publish.

For example, the github hook may get calls for pushes, PRs, issues, etc. which all may go to different topics.